views/admin.functions.php

```diff
@@ -324,3 +324,30 @@ function sailthru_verify_template( $tpl ) {
 
 	return $errors;
 }
+
+// Generate the dropdown for SPM sections. 
+function sailthru_spm_admin_widget( $spm_sections,  $active_section_id, $field_name ) {
+
+	// check to see if we have any sections
+	if ( isset( $spm_sections['sections'] ) &&  count( $spm_sections['sections'] ) > 0 ) {
+		echo '<p>Choose the personalization section to display on your site by selecting the section from the drop down menu below. </p>';
+		echo '<input type="hidden" value="personalize_js" name="sailthru_widget_type" />';
+		echo '<select name="' . $field_name . '">';
+		echo '<option value="">-- Select --</option>';
+
+		foreach ( $spm_sections['sections'] as $section ) {
+
+			if ( $section['section_id'] === $active_section_id ) {
+				$selected = ' selected';
+			} else {
+				$selected = '';
+			}
+			echo  '<option value="' . esc_attr( $section['section_id'] ) . '"' . esc_attr( $selected ) . '>' . esc_attr( $section['name'] ) . '</option>';
```

Comment:
While this is ok, it would be simpler, and eliminate the if else statement to use the `selected` function that comes with WP, additionally the last `esc_attr` can be an `esc_html`:

```
echo  '<option value="' . esc_attr( $section['section_id'] ) . '"' . selected($section['section_id'], $active_section_id ) . '>' . esc_html( $section['name'] ) . '</option>';
```

Otherwise it's good to go as is, these are only optional enhancements :)